openslides_backend/shared/mixins/user_scope_mixin.py

```diff
@@ -160,7 +176,137 @@ def check_permissions_for_scope(
+        """
+        if isinstance(instance_or_id, dict):
+            if not (instance_id := instance_or_id.get("id")):
+                return False
+            if hasattr(self, "name") and self.name == "user.create":
+                if instance_or_id.get("committee_management_ids", []):
+                    return False
+            else:
+                user = self.datastore.get(
+                    fqid_from_collection_and_id("user", instance_id),
+                    ["committee_management_ids"],
+                    lock_result=False,
+                    use_changed_models=False,
+                )
+                if user.get("committee_management_ids", []):
+                    return False
+        else:
+            instance_id = instance_or_id
+        if not b_meeting_ids:
+            if not hasattr(self, "instance_committee_meeting_ids"):
+                return False
+            b_meeting_ids = set()
+            for m_ids in self.instance_committee_meeting_ids.values():
+                if m_ids:
+                    b_meeting_ids.update(m_ids)
+        if not b_meeting_ids:
+            return False
+        # During participant import there is no permstore.
+        if hasattr(self, "permstore"):
+            a_meeting_ids = (
+                self.permstore.user_meetings
+            )  # returns only admin level meetings
+        else:
+            a_user = self.datastore.get(
+                fqid_from_collection_and_id("user", instance_id),
+                ["meeting_ids"],
+                lock_result=False,
+            )
+            a_meeting_ids = set(a_user.get("meeting_ids", []))
+        if not a_meeting_ids:
+            return False
+        intersection_meeting_ids = a_meeting_ids.intersection(b_meeting_ids)
+        if not b_meeting_ids.issubset(intersection_meeting_ids):
+            return False
+        intersection_meetings = self.datastore.get_many(
+            [
+                GetManyRequest(
+                    "meeting",
+                    list(intersection_meeting_ids),
+                    ["admin_group_id", "group_ids", "meeting_user_ids"],
+                )
+            ],
+            lock_result=False,
+        ).get("meeting", {})
+        for meeting_id, meeting_dict in intersection_meetings.items():
+            admin_meeting_users = {}
+            admin_group_id = meeting_dict.get("admin_group_id", 0)
+            admin_group = self.datastore.get(
+                fqid_from_collection_and_id("group", admin_group_id),
+                ["meeting_user_ids"],
+                lock_result=False,
+            )
+            admin_meeting_users = self.datastore.get_many(
+                [
+                    GetManyRequest(
+                        "meeting_user",
+                        admin_group.get("meeting_user_ids", []),
+                        ["user_id"],
+                    )
+                ],
+                lock_result=False,
+            ).get("meeting_user", {})
```

Comment:
May be a good idea to aggregate admin_group_ids across meetings, then load them all with _one_ `get_many` request outside of a loop (again: list comprehensions).
Bonus points for also loading the admin_meeting_users in bulk.

There is also the option of getting just the user with `mapped_fields=["meeting_user_ids"]` and looking for intersections with the `meeting_user_ids` in the admin groups.

Reply:
Done
However I did not look deeper into the latter suggestion. I think, that would mean two n:m comparisons per meeting whereas now it is two 1:n comparisons per meeting.